### PR TITLE
fix: remove wrong candidates 2021-2023 using front-end

### PR DESCRIPTION
**Pull request by drendog**

### CODE REVIEW

File: src/data/DataReader.ts [hunk 1]
```diff
@@ -40,7 +40,40 @@ class DataReader {
...
+          const candidates = this.getCandidates(year, entity, subEntity, list.nome);
+          for (const candidate of candidates) {
+            let isBadCandidate = false;
+            for (const list of lists) {
+              isBadCandidate = isBadCandidate || (list.nome as string).includes(candidate.nominativo);
+            }
+            if (isBadCandidate) {
+              console.log(`FIXED ${entity}-${subEntity}-${list.nome} candidato:${candidate.nominativo}`);
+              this.removeCandidate(year, entity, subEntity, candidate.nominativo);
+            }
```
Comment (Helias) [inline src/data/DataReader.ts:67]: the for-loop iterate even if you already find the element, you could use `if (lists.findI(l -> l.nome == candidate.nominativo)) { console.log('FIXED...'); }`

File: src/data/DataReader.ts [hunk 2]
```diff
@@ -40,7 +40,40 @@ class DataReader {
...
+          }
+        }
+      }
+    }
+  }
+
+  removeCandidate(years: string, entity: string, subEntity: string, candidateName: string): void {
+    this.data[years][entity][subEntity].eletti = (this.data[years][entity][subEntity].eletti as any[]).filter((candidate: any) => !((candidate.nominativo as string).includes(candidateName)));
+    this.data[years][entity][subEntity].non_eletti = (this.data[years][entity][subEntity].non_eletti as any[]).filter((candidate: any) => !((candidate.nominativo as string).includes(candidateName)));
   }
```
Comment (Helias) [inline src/data/DataReader.ts:73]: this is a bit ugly, but it's ok

maybe, we could even create a function that contains:
```typescript
private cleanCandidates(candidateList: string[]): string[] {
  return candidateList
      .filter((candidate: any) => !((candidate.nominativo as string)
      .includes(candidateName));
}

```

or something similar, so removeCandidate can use this just using `this.data[years][entity][subEntity].eletti = cleanCanddiates( this.data[years][entity][subEntity].eletti );`

Comment (Helias) [reply src/data/DataReader.ts:73]: I am not sure about the as any[] and as string

Comment (drendog) [reply src/data/DataReader.ts:73]: it was just a copypaste from getCandidates method

**Pull request merged**